Add Mitsubishi Comfort integration documentation (#44516)
home-assistant/home-assistant.github.io: source/_integrations/mitsubishi_comfort.markdown

---
title: Mitsubishi Comfort
description: Integrate Mitsubishi Electric ductless minisplit heat pump and air conditioning systems with Home Assistant.
ha_category:
  - Climate

> ha_release: 2026.5

Reviewer: ```suggestion
ha_release: 2026.6
```

ha_iot_class: Local Polling
ha_config_flow: true
ha_codeowners:
  - '@nikolairahimi'
ha_domain: mitsubishi_comfort
ha_platforms:
  - climate
ha_integration_type: hub
ha_quality_scale: bronze
---

The **Mitsubishi Comfort** {% term integration %} connects Mitsubishi Electric ductless minisplit heat pump and air conditioning systems to Home Assistant. It communicates directly with each indoor unit over your local network for low-latency control, using the Kumo Cloud account only for initial device discovery and credential retrieval.

This integration supports indoor units (full climate control) and Kumo Station outdoor units (temperature monitoring only).

> 

Reviewer 2: ```suggestion

## Supported devices

- **Indoor unit** (ductless, ducted)
  - Controls: mode, temperature, fan speed, vane direction
  - Sensors: temperature, humidity, filter status, Wi-Fi signal
- **Kumo Station** (outdoor unit, read-only)
  - Sensors: outdoor temperature, Wi-Fi signal

```
We usually keep this section before the prereqs. See https://developers.home-assistant.io/docs/documenting/integration-docs-examples#basic-structure-of-an-integration-page

## Prerequisites

- A [Kumo Cloud](https://app.kumocloud.com) account with your Mitsubishi devices registered
- Devices must be connected to your local network via their Wi-Fi adapters
- During setup you will be asked to enter the **username** and **password** for your Kumo Cloud account. The integration uses these credentials to discover your devices and retrieve the local API passwords needed for direct communication.

{% include integrations/config_flow.md %}

## Supported devices

- **Indoor unit** (ductless, ducted)
  - Controls: mode, temperature, fan speed, vane direction
  - Sensors: temperature, humidity, filter status, Wi-Fi signal
- **Kumo Station** (outdoor unit, read-only)

>   - Sensors: outdoor temperature, Wi-Fi signal

Reviewer 2: ```suggestion
```

## Supported functionality

The **Mitsubishi Comfort** integration provides the following entities.

### Climate

Each indoor unit is exposed as a climate entity with the following capabilities:

- **HVAC modes**: Off, Cool, Heat, Dry, Fan only, Heat/Cool (auto)
  - Availability depends on the specific unit.

> - **Fan modes**: Super Quiet, Quiet, Low, Powerful, Super Powerful, Auto

Reviewer 2: ```suggestion
- **Fan modes**: Super quiet, Quiet, Low, Powerful, Super powerful, Auto
```
We use [sentence-style capitalization](https://learn.microsoft.com/en-us/style-guide/capitalization), also for UI strings.

Author: Got it

This integration supports indoor units (full climate control) and Kumo Station outdoor units (temperature monitoring only).

## Supported devices

- **Indoor unit** (ductless, ducted)
  - Controls: mode, temperature, fan speed, vane direction
  - Sensors: temperature, humidity, filter status, Wi-Fi signal
- **Kumo Station** (outdoor unit, read-only)
  - Sensors: outdoor temperature, Wi-Fi signal

## Prerequisites

- A [Kumo Cloud](https://app.kumocloud.com) account with your Mitsubishi devices registered
- Devices must be connected to your local network via their Wi-Fi adapters
- During setup you will be asked to enter the **username** and **password** for your Kumo Cloud account. The integration uses these credentials to discover your devices and retrieve the local API passwords needed for direct communication.

{% include integrations/config_flow.md %}

> 

Reviewer 2: ```suggestion

{% configuration_basic %}
Username:
  description: The email address for your Kumo Cloud account.
Password:
  description: The password for your Kumo Cloud account.
{% endconfiguration_basic %}
```

---
title: Mitsubishi Comfort

> description: Instructions on how to integrate Mitsubishi minisplit systems into Home Assistant.

Reviewer 2: ```suggestion
description: Integrate Mitsubishi Electric ductless minisplit heat pump and air conditioning systems with Home Assistant.
```

ha_category:
  - Climate
ha_release: 2026.5
ha_iot_class: Local Polling
ha_config_flow: true
ha_codeowners:
  - '@nikolairahimi'
ha_domain: mitsubishi_comfort
ha_platforms:
  - climate

> ha_dhcp: true

Reviewer 2: ```suggestion
```
has this already been implemented? I don't see it in core.

Author: This is a mistake. It was removed from core so shouldn't be here

This integration supports indoor units (full climate control) and Kumo Station outdoor units (temperature monitoring only).

## Prerequisites

- A [Kumo Cloud](https://app.kumocloud.com) account with your Mitsubishi devices registered
- Devices must be connected to your local network via their Wi-Fi adapters
- During setup you will be asked to enter the **username** and **password** for your Kumo Cloud account. The integration uses these credentials to discover your devices and retrieve the local API passwords needed for direct communication.

{% include integrations/config_flow.md %}

After setup, you can adjust the following options under **Settings** > **Devices & services** > **Mitsubishi Comfort** > **Configure**:

- **Connection timeout**: How long to wait when connecting to a device (default: 1.2 seconds)
- **Response timeout**: How long to wait for a device response (default: 8.0 seconds)

> 

Reviewer 2: ```suggestion
```
Are these really user-configurable? In core, it says that there is no Option flow for this integration. https://github.com/home-assistant/core/pull/167472/changes#diff-926cdb920bb74fa46da4ddd810c05d7fe864507d960fdfe3d6234bfeb8446d84R88

> ## Supported devices

Reviewer 2: we usually place this section before the prerequisites. See https://developers.home-assistant.io/docs/documenting/integration-docs-examples#basic-structure-of-an-integration-page

Author: This was removed

> - **Indoor unit** (ductless, ducted) — Full climate control: mode, temperature, fan speed, vane direction. Sensors: temperature, humidity, filter status, Wi-Fi signal.

Reviewer 2: use a nested list for those.

- **Kumo Station** (outdoor unit) — Read-only: outdoor temperature, Wi-Fi signal.

> 

Reviewer 2: ```suggestion
## Supported functionality

The **Mitsubishi Comfort** integration provides the following entities.
```

> ## Climate entity

Reviewer 2: ```suggestion
### Climate
```

Each indoor unit is exposed as a climate entity with the following capabilities:

- **HVAC modes**: Off, Cool, Heat, Dry, Fan Only, Heat/Cool (auto) — availability depends on the specific unit's capabilities
- **Temperature setpoints**: Single setpoint in cool/heat modes, dual setpoints (high/low) in auto mode
- **Fan modes**: Super Quiet, Quiet, Low, Powerful, Super Powerful, Auto — number of speeds depends on the unit

> - **Swing modes**: Horizontal, Mid-Horizontal, Midpoint, Mid-Vertical, Vertical, Auto, Swing — availability depends on the unit

Reviewer 2: ```suggestion
- **HVAC modes**: Off, Cool, Heat, Dry, Fan only, Heat/Cool (auto)
  - Availability depends on the specific unit.
- **Fan modes**: Super Quiet, Quiet, Low, Powerful, Super Powerful, Auto
  - The number of available speeds depends on the unit.
- **Swing modes**: Horizontal, Mid-Horizontal, Midpoint, Mid-Vertical, Vertical, Auto, Swing
  - Availability depends on the unit.
- **Temperature**: Single setpoint in Cool and Heat modes, separate high and low setpoints in Heat/Cool (auto) mode.
  - Temperature ranges vary by unit and mode.
```
I would use a nested list here.

Temperature ranges are read from the device profile and may vary by unit and mode.

## How it works

1. **Setup**: The integration authenticates with Kumo Cloud to discover devices and retrieve per-device API credentials.
2. **Runtime**: All communication happens directly with the devices over your LAN using their local HTTP API. No cloud connection is needed after initial setup.
3. **Polling**: Device status is polled every 60 seconds. Commands (mode changes, temperature adjustments) are sent immediately and use optimistic state updates for responsive UI feedback.

> 4. **Credentials**: Device credentials are cached locally so the integration can reconnect after a restart without contacting the cloud.

Reviewer 2: ```suggestion
```
I would move some of this content into the introduction, and for the polling, we have a dedicated section.

> 

Reviewer 2: ```suggestion
## Data updates

The Mitsubishi Comfort integration {% term polling polls %} the status of your devices every 60 seconds. When you send a command (such as changing the mode or adjusting the temperature), Home Assistant reflects the change straight away, without waiting for the next poll.
```
We have a dedicated section for data updates. Please check if my suggestion is correct.

## DHCP discovery

> The integration can automatically detect Mitsubishi adapters on your network via DHCP (MAC address prefixes `24:CD:8D` and `70:87:A7`). When a device is detected, you will be prompted to set up the integration.

Reviewer 2: ```suggestion
```
Has DHCP already been implemented? I didn't see it in core. Is it maybe planned for a future PR?

---
title: Mitsubishi Comfort
description: Instructions on how to integrate Mitsubishi minisplit systems into Home Assistant.
ha_category:
  - Climate
ha_release: 2026.5
ha_iot_class: Local Polling
ha_config_flow: true
ha_codeowners:
  - '@nikolairahimi'
ha_domain: mitsubishi_comfort
ha_platforms:
  - climate
ha_dhcp: true
ha_integration_type: hub
ha_quality_scale: bronze
---

The **Mitsubishi Comfort** {% term integration %} connects Mitsubishi Electric ductless minisplit heat pump and air conditioning systems to Home Assistant. It communicates directly with each indoor unit over your local network for low-latency control, using the Kumo Cloud account only for initial device discovery and credential retrieval.

This integration supports indoor units (full climate control) and Kumo Station outdoor units (temperature monitoring only).

## Prerequisites

> 
> - A [Kumo Cloud](https://app.kumocloud.com) account with your Mitsubishi devices registered
> - Devices must be connected to your local network via their Wi-Fi adapters
> - During setup you will be asked to enter the **username** and **password** for your Kumo Cloud account. The integration uses these credentials to discover your devices and retrieve the local API passwords needed for direct communication.
> 
> {% include integrations/config_flow.md %}
> 
> 

Reviewer 2: ```suggestion
```

---
title: Mitsubishi Comfort
description: Instructions on how to integrate Mitsubishi minisplit systems into Home Assistant.
ha_category:
  - Climate

> ha_release: "2026.5"

Reviewer 2: ```suggestion
ha_release: 2026.5
```

ha_iot_class: Local Polling
ha_config_flow: true
ha_codeowners:
  - '@nikolairahimi'
ha_domain: mitsubishi_comfort
ha_platforms:
  - climate
ha_dhcp: true
ha_integration_type: hub
ha_quality_scale: bronze
---

> The **Mitsubishi Comfort** integration connects Mitsubishi Electric ductless minisplit heat pump and air conditioning systems to Home Assistant. It communicates directly with each indoor unit over your local network for low-latency control, using the Kumo Cloud account only for initial device discovery and credential retrieval.

Reviewer 2: ```suggestion
The **Mitsubishi Comfort** {% term integration %} connects Mitsubishi Electric ductless minisplit heat pump and air conditioning systems to Home Assistant. It communicates directly with each indoor unit over your local network for low-latency control, using the Kumo Cloud account only for initial device discovery and credential retrieval.
```
There's a [glossary term reference](https://developers.home-assistant.io/docs/documenting/integration-docs-examples#glossary-term-reference) for this term :book: .

This integration supports indoor units (full climate control) and Kumo Station outdoor units (temperature monitoring only).

## Prerequisites

- A [Kumo Cloud](https://app.kumocloud.com) account with your Mitsubishi devices registered

> - Devices must be connected to your local network via their Wi-Fi adapters

Reviewer 2: ```suggestion
- Devices must be connected to your local network via their Wi-Fi adapters
- During setup you will be asked to enter the **username** and **password** for your Kumo Cloud account. The integration uses these credentials to discover your devices and retrieve the local API passwords needed for direct communication.
```
If this is needed during setup, I would list it under the prereqs

{% include integrations/config_flow.md %}

> ## Configuration

Reviewer 2: ```suggestion
```
the title "Configuration" is already part of the include element.

> During setup you will be asked to enter the **username** and **password** for your Kumo Cloud account. The integration uses these credentials to discover your devices and retrieve the local API passwords needed for direct communication.

Reviewer 2: ```suggestion
```
Should this be moved to the prerequisites?

Author: Yeah, I like that more. I'll move it

After setup, you can adjust the following options under **Settings** > **Devices & services** > **Mitsubishi Comfort** > **Configure**:

- **Connection timeout** — How long to wait when connecting to a device (default: 1.2 seconds)

> - **Response timeout** — How long to wait for a device response (default: 8.0 seconds)

Reviewer 2: ```suggestion
- **Connection timeout**: How long to wait when connecting to a device (default: 1.2 seconds)
- **Response timeout**: How long to wait for a device response (default: 8.0 seconds)
```
